### ISSUE TITLE
Autoupload Option for upload when not in roaming

### ISSUE THREAD

**Issue by pahy** [nextcloud/android#583]
Featurerequest:

Option to autoupload the media by lte/gsm when not in roaming mode.

Thank you!

**Comment by AndyScherzinger** [comment]
Hi @pathy,

not sure if I understand your request correctly. If you do want the auto upload to push your stuff to the server while being on a cellular network, then I think you are all set with not checking "wifi only" in the auto upload preferences.
Roaming shouldn't be an issue imho since that should be handled by Android itself.

**Comment by gsantner** [comment]
@AndyScherzinger 
It does make sense to enable roaming for browsing e.g. a website to get some infos when you are in holidays. But you maybe don't want to upload (which will cost more $$$) pictures automatically.

**Comment by pahy** [comment]
Hi @AndyScherzinger, hi @gsantner,

@gsantner you hit the idea precisely :-)
Let me clearify my situation. My contract includes an unlimited lte transfer. Therefore i do not need an alternative transfer by wifi in my home country. But i got a limited transfer volumina for roaming in my contract. As it is expensive to buy respectively to reorder, data intensive transfers in roaming situations are not desired.
I have manually to switch on and off upload by wifi only when crossing the border to gain this effect at the moment.

Thank you!

**Comment by AndyScherzinger** [comment]
@pahy @gsantner to further clarify, that would mean you would cutoff the app completely or just the upload part? Because there would also be browsing the folder structures, retrieving image/video thumbnails, manually downloading/uploading files and "available offline" as in keeping files in sync.

cc: @eppfel @jancborchardt @mario @tobiasKaminsky

**Comment by pahy** [comment]
@AndyScherzinger i am just thinking about a little setting withing the autoupload options saying "upload when roaming"
![roaming](https://cloud.githubusercontent.com/assets/20403536/22377653/13ae4244-e4b3-11e6-980c-9ee641114c99.png)

**Comment by AndyScherzinger** [comment]
But you then still would have to do this for every activated auto upload!

**Comment by pahy** [comment]
You are right. A global setting for it would be great. But setting it individually for each autoupload might also be fine, because i have e.g. an images folder for telegram images. I do not need telegram images or videos uploaded when roaming, but my camera pictures or maybe screenshots.

**Comment by eppfel** [comment]
Are we anyhow making use of the Android system setting "automatically sync data", when I was at FOSDEM these days, I activated roaming, but deactivated this option. I am always for giving users options, but this might solve the problem anyway.

**Comment by pahy** [comment]
Where do i find this option? I got here Android 7 with changes from Huawei (EMUI) and i can't find this option.
As i do not know this option, would it let me sync my calendar (with low data consumption) and stop syncing nextcloud (with high data consumption) when on roaming?

**Comment by eppfel** [comment]
Oh, so maybe this is deprecate or a Sony thing. It basically stops all background syncing: calendar, etc. And you are right, this is different to Auto Upload.

I'd go for the global `pause auto upload`(later: pasue `sync`) and not stop anything else.

**Comment by mario** [comment]
It's significantly easier to add a NO ROAMING option since the job system supports this :)

**Comment by tobiasKaminsky** [comment]
> Are we anyhow making use of the Android system setting "automatically sync data"

This is respected only for a full sync of all folders (fetching all folder and file names), as far as I know.

**Comment by jasonbayton** [comment]
Bump please. Ping @mario @tobiasKaminsky @AndyScherzinger

**Comment by dJani97** [comment]
I would love this feature as well. I have  infinite cellular data in my country, but very limited data when roaming.

So my current Nextcloud setting is to auto-upload all the time, even on cellular. But when I go abroad, I must remember to go into the settings, and set each single folder one-by-one to only auto-upload on WiFi, otherwise it will burn my roaming data.

**Comment by joshtrichards** [comment]
#383 would like be a reasonable approach here to addressing this too
